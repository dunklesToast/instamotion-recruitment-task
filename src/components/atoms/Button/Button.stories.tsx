import React from 'react';
import type { Meta, Story } from '@storybook/react';
import type { ButtonProps } from './Button';
import { Button } from './Button';

export default {
  title: 'Atoms/Button',
  component: Button,
} as Meta;

const ButtonComponent: Story<ButtonProps> = (args) => <Button {...args} />;

export const Primary = ButtonComponent.bind({});
export const Secondary = ButtonComponent.bind({});
Primary.args = {
  children: 'Click me',
  secondary: false,
};

Secondary.args = {
  children: 'Click me',
  secondary: true,
};
