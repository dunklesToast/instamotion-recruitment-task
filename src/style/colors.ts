export const colors = {
  primary: '#0B4256',
  primaryHover: '#0e546d',
  secondary: '#F7B628',
  secondaryHover: '#cb9712',
};
