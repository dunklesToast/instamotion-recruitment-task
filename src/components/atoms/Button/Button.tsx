import type { MouseEvent } from 'react';
import React from 'react';
import { Text } from '@atoms/Text/Text';
import * as S from './Button.styled';

export interface ButtonProps {
  /** Button Text */
  children: React.ReactNode;
  /** Sets to secondary color */
  secondary?: boolean;
  /** onClick Handler */
  onClick?: (e: MouseEvent<HTMLButtonElement>) => void;
  /** Type */
  type?: 'submit' | 'button';
}

/**
 * Description of Button.
 */
export function Button({
  children,
  secondary,
  onClick,
  type = 'button',
}: ButtonProps): JSX.Element {
  return (
    <S.Button secondary={secondary} data-testid='button' onClick={onClick} type={type}>
      <Text>{children}</Text>
    </S.Button>
  );
}
