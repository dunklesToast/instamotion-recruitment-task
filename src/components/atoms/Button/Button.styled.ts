import type { ButtonProps } from '@atoms/Button/Button';
import { colors } from '@style/colors';
import { mediaHover } from '@style/helpers/mixins/mediaHover';
import { untilTablet } from '@style/helpers/mixins/mediaQueries';
import styled, { css } from 'styled-components';

const onHover = mediaHover(css`
  background-color: ${({ secondary }) => (secondary ? colors.secondaryHover : colors.primaryHover)};
  cursor: pointer;
`);

export const Button = styled.button<ButtonProps>`
  background-color: ${({ secondary }) => (secondary ? colors.secondary : colors.primary)};
  border: none;
  border-radius: 8px;
  color: ${({ secondary }) => (secondary ? 'black' : 'white')};
  outline: none;
  padding: 10px 28px;
  transition: background-color 0.1s ease-in-out;

  ${onHover}

  ${untilTablet(css`
    padding: 10px 14px;
  `)}
`;
