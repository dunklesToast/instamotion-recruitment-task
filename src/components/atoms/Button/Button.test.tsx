import React from 'react';
import { colors } from '@style/colors';
import { render, screen } from '@testing-library/react';
import { Button } from './Button';

test('Button', () => {
  const { rerender, asFragment } = render(<Button>Button</Button>);

  const primaryFragment = asFragment();

  const getButton = () => screen.getByTestId('button');

  expect(getButton()).toHaveStyleRule('background-color', colors.primary);
  expect(getButton()).toBeInTheDocument();
  expect(getButton()).toMatchSnapshot();

  rerender(<Button secondary>Button</Button>);

  expect(getButton()).toHaveStyleRule('background-color', colors.secondary);
  expect(getButton()).toBeInTheDocument();
  expect(primaryFragment).toMatchDiffSnapshot(asFragment());
});
