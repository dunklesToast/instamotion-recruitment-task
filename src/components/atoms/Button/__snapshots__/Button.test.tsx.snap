// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button 1`] = `
.c1 {
  font-family: 'PT Sans',sans-serif;
  font-style: regular;
  font-weight: 400;
}

.c0 {
  background-color: #0B4256;
  border: none;
  border-radius: 8px;
  color: white;
  outline: none;
  padding: 10px 28px;
  -webkit-transition: background-color 0.1s ease-in-out;
  transition: background-color 0.1s ease-in-out;
}

@media (hover:hover) {
  .c0:hover {
    background-color: #0e546d;
    cursor: pointer;
  }
}

@media (max-width:1223px) {
  .c0 {
    padding: 10px 14px;
  }
}

<button
  class="c0"
  data-testid="button"
  type="button"
>
  <p
    class="c1"
  >
    Button
  </p>
</button>
`;

exports[`Button 2`] = `
"Snapshot Diff:
- First value
+ Second value

@@ -1,8 +1,8 @@
  <DocumentFragment>
    <button
-     class=\\"sc-gtsrHT eIuphD\\"
+     class=\\"sc-gtsrHT hhjUsd\\"
      data-testid=\\"button\\"
      type=\\"button\\"
    >
      <p
        class=\\"sc-bdnxRM unfAt\\""
`;
